Source/Core/Common/Crypto/HMAC.h

```diff
@@ -0,0 +1,12 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "Common/CommonTypes.h"
+
+namespace Common::HMAC
+{
+bool HMACWithSHA1(const u8* key, const size_t key_len, const u8* msg, const size_t msg_len,
+                  u8* out);
```

Comment:
I assume there's a expected length for the `out` buffer, should probably be mentioned here.

Reply:
Sorry to hijack your comment Admiral.  Could the `key` and `msg` attributes be a `std::span`?

Reply:
To answer iwubcode, yes `std::span` can be used.

For Admiral, there is an expected length yes, would I just make the `out` buffer a `std::array<u8, 20>`?

Reply:
If it makes sense, sure, but I'd be okay if you just documented the length here.